clients/da-vinci-client/src/main/java/com/linkedin/davinci/consumer/VeniceChangelogConsumerImpl.java

```diff
@@ -219,7 +220,7 @@ public CompletableFuture<Void> internalSubscribe(Set<Integer> partitions, PubSub
       }
 
       synchronized (pubSubConsumer) {
-        Set<PubSubTopicPartition> topicPartitionSet = new HashSet<>(pubSubConsumer.getAssignment());
+        Set<PubSubTopicPartition> topicPartitionSet = getTopicAssignment();
```

Comment:
We need lock on `topicPartitionSet` here?

Reply:
I don't think so.  We're already synchronized on pubSubConsumer and the synchronizedSet to boot won't throw the exception.  I think we're safe.

Reply:
The previous synchronized is for preventing the internal new HashSet<> concurrent exception issue, but now its not needed as you already added the check inside `getTopicAssignment`. But for the synchronizedSet, we need to do a locks on the returned object. https://docs.oracle.com/javase/8/docs/api/java/util/Collections.html#synchronizedSet-java.util.Set-

Reply:
The above is speaking to the behavior of iteration on this set if the contents change during the iteration.  We're syncrhonizing on the pubsubconsumer.  If we synchronize on the pubsub consumer, I don't know how this collection can change.

Reply:
My point was we dont need to synchronize on the pubsub consumer given `getTopicAssignment()` is doing it now already.

Reply:
Right, but the rest of the code block does need the synchronization because we're changing the assignment.  I guess we could shrink it if we want?